include/ddc/kernels/splines/spline_boundary_conditions.hpp

```diff
@@ -8,16 +8,22 @@
 #include <stdexcept>
 
 namespace ddc {
+
+/// @brief An enum representing a spline boundary condition.
 enum class BoundCond {
-    // Periodic boundary condition u(1)=u(n)
-    PERIODIC,
-    // Hermite boundary condition
-    HERMITE,
-    // Use Greville points instead of conditions on derivative for B-Spline
-    // interpolation
-    GREVILLE,
+    PERIODIC, ///< Periodic boundary condition u(1)=u(n)
+    HERMITE, ///< Hermite boundary condition
+    GREVILLE, ///< Use Greville points instead of conditions on derivative for B-Spline interpolation
```

Comment:
I think for these two boundary conditions we will need at least a reference to explain what they are ?

Reply:
In #437 there is a link to @EmilyBourne thesis. Maybe we can write it there too ? Or Emily has another proposal ?

Reply:
Yes seems good to me